tests/sentry/integrations/jira_server/test_integration.py

```diff
@@ -810,8 +804,94 @@ def test_sync_assignee_outbound_no_email(self):
+        integration = self.create_integration(
+            organization=self.organization,
+            provider="jira_server",
+            external_id="jira:1",
+            name="Example Jira",
+            oi_params={
+                "config": {
+                    "sync_comments": True,
+                    "sync_forward_assignment": True,
+                    "sync_reverse_assignment": True,
+                    "sync_status_reverse": True,
+                    "sync_status_forward": True,
+                }
+            },
+        )
+        self.create_integration_external_project(
+            organization_id=self.organization.id,
+            integration_id=integration.id,
+            external_id="12345",
+            unresolved_status="in_progress",
+            resolved_status="done",
+        )
+
+        installation = integration.get_installation(self.organization.id)
+
+        assert installation.get_config_data() == {
+            "sync_comments": True,
+            "sync_forward_assignment": True,
+            "sync_reverse_assignment": True,
+            "sync_status_reverse": True,
+            "sync_status_forward": {"12345": {"on_resolve": "done", "on_unresolve": "in_progress"}},
+            "issues_ignored_fields": "",
+        }
+
+    def test_get_config_data_issues_keys(self):
+        integration, org_integration = self.create_provider_integration_for(
+            organization=self.organization,
+            user=self.user,
+            provider="jira_server",
+            name="Example Jira",
+        )
+        installation = integration.get_installation(self.organization.id)
+
+        # If config has not be configured yet, uses empty string fallback
+        assert "issues_ignored_fields" not in org_integration.config
+        assert installation.get_config_data().get("issues_ignored_fields") == ""
+
+        # List is serialized as comma-separated list
+        with assume_test_silo_mode(SiloMode.CONTROL):
+            org_integration.config["issues_ignored_fields"] = ["hello world", "goodnight", "moon"]
+            org_integration.save()
+        installation = integration.get_installation(self.organization.id)
+        assert (
+            installation.get_config_data().get("issues_ignored_fields")
+            == "hello world, goodnight, moon"
+        )
+
+    def test_create_comment(self):
+        group_note = mock.Mock()
+        comment = "hello world\nThis is a comment.\n\n\n    Glad it's quoted"
+        group_note.data = {"text": comment}
+        with mock.patch.object(StubJiraApiClient, "create_comment") as mock_create_comment:
+            with mock.patch.object(self.installation, "get_client", get_client):
+                self.installation.create_comment(1, self.user.id, group_note)
+                assert (
+                    mock_create_comment.call_args[0][1]
+                    == "Sentry Admin wrote:\n\n{quote}%s{quote}" % comment
+                )
```

Comment:
Could we handle this with a responses mock? Right now the tests are tightly coupled to how the code is written. We care about the outbound request that is being made which is what responses would let us ensure.

Reply:
I'll merge this as is and begin on a new PR for removing this stubbing, I agree and it seems like a much bigger change once I started poking at it.